src/Serilog.Sinks.Graylog/MessageBuilders/GelfMessageBuilder.cs

```
@@ -16,22 +15,21 @@ namespace Serilog.Sinks.Graylog.MessageBuilders
     /// <seealso cref="Serilog.Sinks.Graylog.MessageBuilders.IMessageBuilder" />
     public class GelfMessageBuilder : IMessageBuilder
     {
-        
         private readonly string _hostName;
-        private readonly IPropertyNamingStrategy _propertyNamingStrategy;
         private const string GelfVersion = "1.1";
+        private readonly IPropertyNamingStrategy _propertyNamingStrategy;
         protected GraylogSinkOptions Options { get; }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="GelfMessageBuilder"/> class.
         /// </summary>
         /// <param name="hostName">Name of the host.</param>
         /// <param name="options">The options.</param>
-        public GelfMessageBuilder(string hostName, GraylogSinkOptions options)
+        public GelfMessageBuilder(string hostName = null, GraylogSinkOptions options = null)
         {
-            _hostName = hostName;
```

This throws an NRE at log time below if null is provided on dependencies. Setting reasonable defaults in constructor, alternatively, throw arg exceptions and fail fast.

---

@whir1 can we get this merged ?

---

nope =D

---

Cool, I'm sure you can manage to fix it yourself =D
For the record, its not acceptable to "refactor" your peers contributions and leave their work in a non functioning state. I can't think of a better way to lose contributor support. Your latest changes completely broke UDP as well as the field new naming conventions, which would have been quite obvious if they were tested.

---

Hmmm. Yes i can. But i think that your changes is not a good functionality to this sink. Especialy PropertyNaming =\. And i think to delete it.